gamemodes/terrortown/entities/entities/base_ammo_ttt.lua

```diff
@@ -135,42 +118,61 @@ function ENT:CheckForWeapon(ply)
 end
 
 ---
--- @param Entity ent
+-- This entity function is called once another entity toruches it.
+-- @param Entity ply The touching entity that is probably a player
 -- @realm shared
-function ENT:Touch(ent)
-	if SERVER and not self.tickRemoval and ent:IsValid() and ent:IsPlayer() and self:CheckForWeapon(ent) and self:PlayerCanPickup(ent) then
-		local ammo = ent:GetAmmoCount(self.AmmoType)
+function ENT:Touch(ply)
+	if CLIENT or self.tickRemoval or not ply:IsValid() or not ply:IsPlayer()
+		or not self:CheckForWeapon(ply) or not self:PlayerCanPickup(ply)
+		---
+		-- @realm shared
+		or hook.Run("TTTCanPickupAmmo", ply, self) == false
```

Comment:
Before, returning `true` in `TTTCanPickupAmmo` followed in the result that `Player:PlayerCanPickup()` returned `true` as well. Now you changed the behavior. Theoretically, you just have to check above if the return is not `nil` instead of removing it there.

Reply:
Doesn't change anything about the above mentioned. It doesn't makes sense if PlayerCanPickup returns true, but the hook is preventing it still.